scripts/ssl_tts/ssl_tts_vc.py

```diff
@@ -0,0 +1,254 @@
+
+import librosa
+import soundfile
+import torch
+
+from nemo.collections.asr.parts.preprocessing.features import WaveformFeaturizer
+from nemo.collections.tts.models import fastpitch_ssl, hifigan, ssl_tts
+from nemo.collections.tts.torch.helpers import get_base_dir
+
+
+def load_wav(wav_path, wav_featurizer, pad_multiple=1024):
+    wav = wav_featurizer.process(wav_path)
+    if wav.shape[0 % pad_multiple] != 0:
+        wav = torch.cat([wav, torch.zeros(pad_multiple - wav.shape[0] % pad_multiple, dtype=torch.float)])
+    wav = wav[:-1]
+
+    return wav
+
+
+def get_pitch_contour(wav, pitch_mean=None, pitch_std=None, compute_mean_std=False):
+    f0, _, _ = librosa.pyin(
+        wav.numpy(),
+        fmin=librosa.note_to_hz('C2'),
+        fmax=librosa.note_to_hz('C7'),
+        frame_length=1024,
+        hop_length=256,
+        sr=22050,
+        center=True,
+        fill_na=0.0,
+    )
+    pitch_contour = torch.tensor(f0, dtype=torch.float32)
+    _pitch_mean = pitch_contour.mean().item()
+    _pitch_std = pitch_contour.std().item()
+    if compute_mean_std:
+        pitch_mean = _pitch_mean
+        pitch_std = _pitch_std
+    if (pitch_mean is not None) and (pitch_std is not None):
+        pitch_contour = pitch_contour - pitch_mean
+        pitch_contour[pitch_contour == -pitch_mean] = 0.0
+        pitch_contour = pitch_contour / pitch_std
+
+    return pitch_contour
+
+
+def segment_wav(wav, segment_length=44100, hop_size=44100, min_segment_size=22050):
+    if len(wav) < segment_length:
+        pad = torch.zeros(segment_length - len(wav))
+        segment = torch.cat([wav, pad])
+        return [segment]
+    else:
+        si = 0
+        segments = []
+        while si < len(wav) - min_segment_size:
+            segment = wav[si : si + segment_length]
+            if len(segment) < segment_length:
+                pad = torch.zeros(segment_length - len(segment))
+                segment = torch.cat([segment, pad])
+
+            segments.append(segment)
+            si += hop_size
+        return segments
+
+
+def get_speaker_embedding(ssl_model, wav_featurizer, audio_paths, duration=None, device="cpu"):
+    all_segments = []
+    all_wavs = []
+    for audio_path in audio_paths:
+        wav = load_wav(audio_path, wav_featurizer)
+        segments = segment_wav(wav)
+        all_segments += segments
+        all_wavs.append(wav)
+        if duration is not None and len(all_segments) >= duration:
+            # each segment is 2 seconds with one second overlap.
+            # so 10 segments would mean 0 to 2, 1 to 3.. 9 to 11 (11 seconds.)
+            all_segments = all_segments[: int(duration)]
+            break
+
+    signal_batch = torch.stack(all_segments)
+    print("signal batch", signal_batch.shape)
+    # print("signal batch", signal_batch.shape)
+    signal_length_batch = torch.stack([torch.tensor(signal_batch.shape[1]) for _i in range(len(all_segments))])
+    # print("signal length", signal_length_batch.shape)
```

Comment:
Remove commented out prints. (Is the top one supposed to be there too?)

Reply:
Removed the print statement and the commented out ones.